### DISCUSSION TITLE
Open API for models

### DISCUSSION THREAD

**Discussion by ewitkon** [microsoft/semantic-kernel#1986] [General]
If I want my semantic kernel to connect to my model which is hosted on my server, what interface does my model have to expose so it can be supported by semantic kernel?
Do we need to contribute something to this project similar to `kernel.add_text_completion_service("dv", AzureTextCompletion("text-davinci-003", endpoint, api_key))`?

**Comment by evchaki** [comment]
@ewitkon  you will want to mimic a connector like these - https://github.com/microsoft/semantic-kernel/tree/main/dotnet/src/Connectors/Connectors.OpenAI
